5.1.2: test suite is failing

Using tox:
```
[tkloczko@barrel sphinxcontrib-spelling-5.1.2]$ tox --skip-missing-interpreters
py38 create: /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/.tox/py38
py38 installdeps: .[test]
py38 develop-inst: /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2
py38 installed: alabaster==0.7.12,attrs==19.3.0,Babel==2.8.0,certifi==2020.6.20,chardet==3.0.4,coverage==5.2.1,docutils==0.16,extras==1.0.0,fixtures==3.0.0,idna==2.10,imagesize==1.2.0,Jinja2==2.11.2,linecache2==1.0.0,MarkupSafe==1.1.1,more-itertools==8.4.0,packaging==20.4,pbr==5.4.5,pluggy==0.13.1,py==1.9.0,pyenchant==3.1.1,Pygments==2.6.1,pyparsing==2.4.7,pytest==5.4.3,pytest-cov==2.10.0,python-mimeparse==1.6.0,pytz==2020.1,requests==2.24.0,six==1.15.0,snowballstemmer==2.0.0,Sphinx==3.1.2,sphinxcontrib-applehelp==1.0.2,sphinxcontrib-devhelp==1.0.2,sphinxcontrib-htmlhelp==1.0.3,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.3,sphinxcontrib-serializinghtml==1.1.4,# Editable install with no version control (sphinxcontrib-spelling==5.1.2),-e /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2,testtools==2.4.0,traceback2==1.4.0,unittest2==1.1.0,urllib3==1.25.10,wcwidth==0.2.5
py38 run-test-pre: PYTHONHASHSEED='4263156227'
py38 run-test: commands[0] | pytest -v --cov=sphinxcontrib.spelling --cov-report term-missing --cov-config /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/.coveragerc
========================================================================================= test session starts ==========================================================================================
platform linux -- Python 3.8.3, pytest-5.4.3, py-1.9.0, pluggy-0.13.1 -- /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/.tox/py38/bin/python
cachedir: .tox/py38/.pytest_cache
rootdir: /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2
plugins: cov-2.10.0
collected 15 items

sphinxcontrib/spelling/tests/test_builder.py::test_setup PASSED                                                                                                                                  [  6%]
sphinxcontrib/spelling/tests/test_builder.py::test_title PASSED                                                                                                                                  [ 13%]
sphinxcontrib/spelling/tests/test_builder.py::test_body PASSED                                                                                                                                   [ 20%]
sphinxcontrib/spelling/tests/test_builder.py::test_ignore_literals PASSED                                                                                                                        [ 26%]
sphinxcontrib/spelling/tests/test_builder.py::test_several_word_lists PASSED                                                                                                                     [ 33%]
sphinxcontrib/spelling/tests/test_checker.py::test_errors_only PASSED                                                                                                                            [ 40%]
sphinxcontrib/spelling/tests/test_checker.py::test_with_suggestions PASSED                                                                                                                       [ 46%]
sphinxcontrib/spelling/tests/test_checker.py::test_with_wordlist PASSED                                                                                                                          [ 53%]
sphinxcontrib/spelling/tests/test_checker.py::test_with_context_line PASSED                                                                                                                      [ 60%]
sphinxcontrib/spelling/tests/test_checker.py::test_line_of_index_one_line PASSED                                                                                                                 [ 66%]
sphinxcontrib/spelling/tests/test_checker.py::test_line_of_index_multi_line PASSED                                                                                                               [ 73%]
sphinxcontrib/spelling/tests/test_filter.py::test_builtin_unicode PASSED                                                                                                                         [ 80%]
sphinxcontrib/spelling/tests/test_filter.py::test_builtin_regular PASSED                                                                                                                         [ 86%]
sphinxcontrib/spelling/tests/test_filter.py::test_acronym PASSED                                                                                                                                 [ 93%]
sphinxcontrib/spelling/tests/test_filter.py::test_acronym_unicode PASSED                                                                                                                         [100%]

=========================================================================================== warnings summary ===========================================================================================
sphinxcontrib/spelling/filters.py:9
  /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/sphinxcontrib/spelling/filters.py:9: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

.tox/py38/lib/python3.8/site-packages/unittest2/compatibility.py:143
  /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/.tox/py38/lib/python3.8/site-packages/unittest2/compatibility.py:143: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    class ChainMap(collections.MutableMapping):

sphinxcontrib/spelling/tests/test_builder.py::test_setup
sphinxcontrib/spelling/tests/test_builder.py::test_title
sphinxcontrib/spelling/tests/test_builder.py::test_body
sphinxcontrib/spelling/tests/test_builder.py::test_ignore_literals
sphinxcontrib/spelling/tests/test_builder.py::test_several_word_lists
  /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/sphinxcontrib/spelling/checker.py:24: DeprecationWarning: passing 'filters' as a non-keyword argument to get_tokenizer() is deprecated
    self.tokenizer = get_tokenizer(tokenizer_lang, filters)

-- Docs: https://docs.pytest.org/en/latest/warnings.html

----------- coverage: platform linux, python 3.8.3-final-0 -----------
Name                                  Stmts   Miss  Cover   Missing
-------------------------------------------------------------------
sphinxcontrib/spelling/__init__.py       23      1    96%   14
sphinxcontrib/spelling/builder.py       121     13    89%   57-58, 68, 90-95, 134, 145, 150, 179
sphinxcontrib/spelling/checker.py        38      1    97%   34
sphinxcontrib/spelling/directive.py      22     12    45%   32-51
sphinxcontrib/spelling/filters.py        65     19    71%   32-33, 36-39, 123, 127, 131, 141-142, 145, 151, 154, 161-162, 171-172, 189
-------------------------------------------------------------------
TOTAL                                   269     46    83%

==================================================================================== 15 passed, 7 warnings in 0.99s ====================================================================================
linter create: /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/.tox/linter
linter installdeps: .[linter]
linter develop-inst: /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2
linter installed: alabaster==0.7.12,Babel==2.8.0,certifi==2020.6.20,chardet==3.0.4,docutils==0.16,flake8==3.8.2,idna==2.10,imagesize==1.2.0,Jinja2==2.11.2,MarkupSafe==1.1.1,mccabe==0.6.1,packaging==20.4,pycodestyle==2.6.0,pyenchant==3.1.1,pyflakes==2.2.0,Pygments==2.6.1,pyparsing==2.4.7,pytz==2020.1,requests==2.24.0,six==1.15.0,snowballstemmer==2.0.0,Sphinx==3.1.2,sphinxcontrib-applehelp==1.0.2,sphinxcontrib-devhelp==1.0.2,sphinxcontrib-htmlhelp==1.0.3,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.3,sphinxcontrib-serializinghtml==1.1.4,# Editable install with no version control (sphinxcontrib-spelling==5.1.2),-e /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2,urllib3==1.25.10
linter run-test-pre: PYTHONHASHSEED='4263156227'
linter run-test: commands[0] | flake8 sphinxcontrib setup.py
sphinxcontrib/spelling/tests/test_builder.py:12:1: F401 'os' imported but unused
import os
^
sphinxcontrib/spelling/tests/test_builder.py:15:1: F401 'fixtures' imported but unused
import fixtures
^
sphinxcontrib/spelling/tests/test_builder.py:48:1: E302 expected 2 blank lines, found 1
def test_setup(sphinx_project):
^
sphinxcontrib/spelling/tests/test_checker.py:9:1: F401 'unittest' imported but unused
import unittest
^
sphinxcontrib/spelling/tests/test_filter.py:8:1: F401 'unittest' imported but unused
import unittest
^
ERROR: InvocationError for command /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/.tox/linter/bin/flake8 sphinxcontrib setup.py (exited with code 1)
docs create: /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/.tox/docs
docs installdeps: .[test]
docs develop-inst: /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2
docs installed: alabaster==0.7.12,attrs==19.3.0,Babel==2.8.0,certifi==2020.6.20,chardet==3.0.4,coverage==5.2.1,docutils==0.16,extras==1.0.0,fixtures==3.0.0,idna==2.10,imagesize==1.2.0,Jinja2==2.11.2,linecache2==1.0.0,MarkupSafe==1.1.1,more-itertools==8.4.0,packaging==20.4,pbr==5.4.5,pluggy==0.13.1,py==1.9.0,pyenchant==3.1.1,Pygments==2.6.1,pyparsing==2.4.7,pytest==5.4.3,pytest-cov==2.10.0,python-mimeparse==1.6.0,pytz==2020.1,requests==2.24.0,six==1.15.0,snowballstemmer==2.0.0,Sphinx==3.1.2,sphinxcontrib-applehelp==1.0.2,sphinxcontrib-devhelp==1.0.2,sphinxcontrib-htmlhelp==1.0.3,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-qthelp==1.0.3,sphinxcontrib-serializinghtml==1.1.4,# Editable install with no version control (sphinxcontrib-spelling==5.1.2),-e /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2,testtools==2.4.0,traceback2==1.4.0,unittest2==1.1.0,urllib3==1.25.10,wcwidth==0.2.5
docs run-test-pre: PYTHONHASHSEED='4263156227'
docs run-test: commands[0] | sphinx-build -W -b html docs/source docs/build
Running Sphinx v3.1.2
Initializing Spelling Checker
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 6 source files that are out of date
updating environment: [new config] 6 added, 0 changed, 0 removed
reading sources... [ 16%] customize
reading sources... [ 33%] developers
reading sources... [ 50%] history
reading sources... [ 66%] index
reading sources... [ 83%] install
reading sources... [100%] run

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 16%] customize
writing output... [ 33%] developers
writing output... [ 50%] history
writing output... [ 66%] index
writing output... [ 83%] install
writing output... [100%] run

generating indices...  genindexdone
writing additional pages...  searchdone
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in docs/build.
docs run-test: commands[1] | sphinx-build -b spelling docs/source docs/build
Running Sphinx v3.1.2
Initializing Spelling Checker
loading pickled environment... done
Ignoring wiki words
Ignoring acronyms
Adding package names from PyPI to local dictionary…
Ignoring Python builtins
Ignoring importable module names
Adding contents of /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/docs/source/spelling_wordlist.txt to custom word list
Adding contents of /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/docs/source/spelling_people.txt to custom word list
Looking for custom word list in /tmp/tmp64m16q4x/spelling_wordlist.txt
building [mo]: targets for 0 po files that are out of date
building [spelling]: all documents
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
preparing documents... done
writing output... [ 16%] customize
writing output... [ 33%] developers
writing output... [ 50%] history
history.rst:23:Hong:["Hon", "Hmong", "Song", "Tong", "Long", "Dong", "Hung", "Gong", "Pong", "Bong", "Cong", "Yong", "Kong", "Wong"]:Fixed detection of builtins under PyPy, contributed by Hong Minhee
writing output... [ 66%] index
writing output... [ 83%] install
writing output... [100%] run

Spelling checker messages written to /home/tkloczko/rpmbuild/BUILD/sphinxcontrib-spelling-5.1.2/docs/build/output.txt
WARNING: Found 1 misspelled words
build succeeded, 1 warning.
_______________________________________________________________________________________________ summary ________________________________________________________________________________________________
  py38: commands succeeded
ERROR:   linter: commands failed
  docs: commands succeeded
```
Or usimg nose:
```
+ cd sphinxcontrib-spelling-5.1.2
+ /usr/bin/python3 -m nose
EEEEE..........
======================================================================
ERROR: sphinxcontrib.spelling.tests.test_builder.test_setup
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
TypeError: test_setup() missing 1 required positional argument: 'sphinx_project'

======================================================================
ERROR: sphinxcontrib.spelling.tests.test_builder.test_title
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
TypeError: test_title() missing 1 required positional argument: 'sphinx_project'

======================================================================
ERROR: sphinxcontrib.spelling.tests.test_builder.test_body
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
TypeError: test_body() missing 1 required positional argument: 'sphinx_project'

======================================================================
ERROR: sphinxcontrib.spelling.tests.test_builder.test_ignore_literals
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
TypeError: test_ignore_literals() missing 1 required positional argument: 'sphinx_project'

======================================================================
ERROR: sphinxcontrib.spelling.tests.test_builder.test_several_word_lists
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
TypeError: test_several_word_lists() missing 1 required positional argument: 'sphinx_project'

----------------------------------------------------------------------
Ran 15 tests in 0.327s

FAILED (errors=5)
```
BTW: which one method should be used?

--- a/.travis.yml
+++ b/.travis.yml
@@ -26,16 +26,16 @@ before_install:
   - sudo apt-get -qq update
   - sudo apt-get install -y enchant
   - pip uninstall -y nose mock
-  - pip install pbr tox tox-travis
 
 install:
+  - pip install pbr tox tox-travis
   - pip install .[test] .[linter]
 
 script:
   # Conditional builds (related to env passed)
   - |
     case ${BUILD} in
-        docs|pep8|cover)
+        docs|pep8|cover|linter)
             tox -e ${BUILD}
             ;;
         *)

--- a/sphinxcontrib/spelling/tests/test_builder.py
+++ b/sphinxcontrib/spelling/tests/test_builder.py
@@ -9,10 +9,7 @@ import pytest
 
 import codecs
 import io
-import os
 import textwrap
-
-import fixtures
 
 from sphinx.application import Sphinx
 
@@ -44,6 +41,7 @@ def get_sphinx_output(srcdir, outdir):
     with codecs.open(app.builder.output_filename, 'r') as f:
         output_text = f.read()
     return (stdout, stderr, output_text)
+
 
 def test_setup(sphinx_project):
     srcdir, outdir = sphinx_project

--- a/sphinxcontrib/spelling/tests/test_checker.py
+++ b/sphinxcontrib/spelling/tests/test_checker.py
@@ -6,7 +6,6 @@
 """
 
 import os
-import unittest
 
 from sphinxcontrib.spelling.checker import SpellingChecker, line_of_index
 

--- a/sphinxcontrib/spelling/tests/test_filter.py
+++ b/sphinxcontrib/spelling/tests/test_filter.py
@@ -5,8 +5,6 @@
 """Tests for filters.
 """
 
-import unittest
-
 from enchant.tokenize import get_tokenizer
 
 from sphinxcontrib.spelling import filters

